exchanges/bybit/bybit_websocket.go

```diff
@@ -241,203 +241,211 @@ func (by *Bybit) WsDataHandler() {
+				err := json.Unmarshal(respRaw, &data)
+				if err != nil {
+					return err
+				}
 
-			return trade.AddTradesToBuffer(by.Name, trade.Data{
-				Timestamp:    data.TradeData.Time.Time(),
-				CurrencyPair: p,
-				AssetType:    asset.Spot,
-				Exchange:     by.Name,
-				Price:        data.TradeData.Price,
-				Amount:       data.TradeData.Size,
-				Side:         side,
-				TID:          data.TradeData.ID,
-			})
-		case wsTicker:
-			var data WsSpotTicker
-			err := json.Unmarshal(respRaw, &data)
-			if err != nil {
-				return err
-			}
+				p, err := by.extractCurrencyPair(data.Kline.Symbol, asset.Spot)
+				if err != nil {
+					return err
+				}
 
-			p, err := by.extractCurrencyPair(data.Ticker.Symbol, asset.Spot)
-			if err != nil {
-				return err
+				by.Websocket.DataHandler <- stream.KlineData{
+					Pair:       p,
+					AssetType:  asset.Spot,
+					Exchange:   by.Name,
+					StartTime:  data.Kline.StartTime.Time(),
+					Interval:   data.Parameters.KlineType,
+					OpenPrice:  data.Kline.OpenPrice,
+					ClosePrice: data.Kline.ClosePrice,
+					HighPrice:  data.Kline.HighPrice,
+					LowPrice:   data.Kline.LowPrice,
+					Volume:     data.Kline.Volume,
+				}
+				return nil
+			default:
+				by.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: by.Name + stream.UnhandledMessage + string(respRaw)}
 			}
+		}
 
-			by.Websocket.DataHandler <- &ticker.Price{
-				ExchangeName: by.Name,
-				Bid:          data.Ticker.Bid,
-				Ask:          data.Ticker.Ask,
-				LastUpdated:  data.Ticker.Time.Time(),
-				AssetType:    asset.Spot,
-				Pair:         p,
-			}
+		if m, ok := d["auth"]; ok {
+			log.Infof(log.WebsocketMgr, "%v received auth response: %v", by.Name, m)
 			return nil
-		case wsKlines:
-			var data KlineStream
-			err := json.Unmarshal(respRaw, &data)
-			if err != nil {
-				return err
-			}
-
-			p, err := by.extractCurrencyPair(data.Kline.Symbol, asset.Spot)
-			if err != nil {
-				return err
-			}
+		}
 
-			by.Websocket.DataHandler <- stream.KlineData{
-				Pair:       p,
-				AssetType:  asset.Spot,
-				Exchange:   by.Name,
-				StartTime:  data.Kline.StartTime.Time(),
-				Interval:   data.Parameters.KlineType,
-				OpenPrice:  data.Kline.OpenPrice,
-				ClosePrice: data.Kline.ClosePrice,
-				HighPrice:  data.Kline.HighPrice,
-				LowPrice:   data.Kline.LowPrice,
-				Volume:     data.Kline.Volume,
-			}
+		if m, ok := d["ping"]; ok {
+			log.Infof(log.WebsocketMgr, "%v received ping: %v", by.Name, m)
 			return nil
-		default:
-			by.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: by.Name + stream.UnhandledMessage + string(respRaw)}
 		}
-	}
-
-	if m, ok := multiStreamData["auth"]; ok {
-		log.Infof(log.WebsocketMgr, "%v received auth response: %v", by.Name, m)
-		return nil
-	}
-
-	if m, ok := multiStreamData["ping"]; ok {
-		log.Infof(log.WebsocketMgr, "%v received ping: %v", by.Name, m)
-		return nil
+	case []interface{}:
+		if obj, ok := d[0].(map[string]interface{}); ok {
```

Comment:
I think this would be better as a for loop. Removes the need for a len check as well
```go 
for i := range d {
   obj, ok := d[i].(map[string]interface{})
   if !ok {
      return nil, common.GetAssertError("map[string]interface{}", d[i])
   }
   e, ok := obj["e"].(string)
   if !ok {
      return nil, common.GetAssertError("string", obj["e"].(string))
   }

...

}

```